mainscripts/Trainer.py

```diff
@@ -292,30 +307,101 @@ def main(args, device_args):
             key_events = io.get_key_events(wnd_name)
             key, chr_key, ctrl_pressed, alt_pressed, shift_pressed = key_events[-1] if len(key_events) > 0 else (0,0,False,False,False)
             
+            #Gan-er comment:
+            #Change the zooming-in to taste. 
+            #Add remove change the values as you see fit. 
+            #The numbers in the show_last_history_iters_count simply tell you how many updates you want to see.
+            #So for example, 50 means the last 50 updates. All the previous ones will be pruned. 
+            #If you want to change the 50 to e.g. 75 go ahead and change the two '50' numbers below to '75', and so on.
+            #I will update this to be more sophisticated once I have the time. 
+            #The truth is that for a correct zooming implementation (where not just the last X updates are visible)
+            #but also the scale of the y-axis in the preview window changes the entire thing has to be refactored/converted
+            #using an MVC architecture. As it is now things that have to do with the presentation are intermingled with
+            #things that have to do with the modeling without any controllers so the whole thing is messy to deal with.
+            #Note that you need to be able to zoom-in/out in order to evaluate the model's effectiveness especially if you are
+            #doing more advanced stuff like warm restarts and what not (which you should be doing). For more details on wth 
+            #I am talking about you should check the wiki. [which at this moment does not exist] 
+            
+            #G---START
             if key == ord('\n') or key == ord('\r'):
+                s2c.put ( {'op': 'save'} )
                 s2c.put ( {'op': 'close'} )
-            elif key == ord('s'):
+            elif key == ord('s') or key ==ord('S'):
```

Comment:
I don't see much of a point to adding capitalization to certain keys, I don't see why someone would press shift randomly in the window when a lower key is already present.

Reply:
Caps lock is frequently and accidentally pressed. Also I do not understand what is the complain/concern here. It costs nothing to add 'or' clauses. In fact, perhaps cases for cyrillic, greek, etc could be added.